.circleci/config.yml

```diff
@@ -0,0 +1,170 @@
+          # Copy files from teplates
+          cp project-base/app/config/domains_urls.yml.dist project-base/app/config/domains_urls.yml
+          cp project-base/app/config/parameters_test.yml.dist project-base/app/config/parameters_test.yml
+          cp project-base/app/config/parameters.yml.dist project-base/app/config/parameters.yml
+          # Add domains into domain file
+          yq write --inplace project-base/app/config/domains_urls.yml domains_urls[0].url http://$FIRST_DOMAIN_HOSTNAME
+          yq write --inplace project-base/app/config/domains_urls.yml domains_urls[1].url http://$SECOND_DOMAIN_HOSTNAME
+
+    - run:
+        name: Set SSH credentials
+        command: |
+          echo -e $SERVER_SSL_CERT > ~/shopsys
+          chmod 0400 ~/shopsys
+
+    - run:
+        name: Upload repository into cluster, create volumes and set correct permissions
+        command: |
+          cd .. && tar -zcvf source.tar.gz source
+          ssh -o "StrictHostKeyChecking no" -i ~/shopsys www-data@heimdall.shopsys.com mkdir -p /workspace/shopsys/$CIRCLE_BRANCH
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com rm -rf /workspace/shopsys/$CIRCLE_BRANCH/source
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com sudo chown -R www-data:www-data /workspace/shopsys/$CIRCLE_BRANCH/db
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com rm -rf /workspace/shopsys/$CIRCLE_BRANCH/db/*
+          scp -i ~/shopsys source.tar.gz www-data@heimdall.shopsys.com:/workspace/shopsys/$CIRCLE_BRANCH/source.tar.gz
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com tar -zxvf /workspace/shopsys/$CIRCLE_BRANCH/source.tar.gz -C /workspace/shopsys/$CIRCLE_BRANCH/
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com rm /workspace/shopsys/$CIRCLE_BRANCH/source.tar.gz
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com mkdir -p /workspace/shopsys/$CIRCLE_BRANCH/db
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com chown -R www-data:www-data /workspace/shopsys/$CIRCLE_BRANCH
+
+    - run:
+        name: Deploy application to Kubernetes cluster
+        command: |
+          kubectl delete namespace $CIRCLE_BRANCH || true
+          kubectl create namespace $CIRCLE_BRANCH || true
+          kubectl config set-context $(kubectl config current-context) --namespace=$CIRCLE_BRANCH
+          kubectl apply --recursive -f .
+        working_directory: project-base/k8s
+
+    - run:
+        name: Wait for rollouts
+        command: |
+          kubectl rollout status deployment/adminer --watch
+          kubectl rollout status deployment/postgres --watch
+          kubectl rollout status deployment/redis --watch
+          kubectl rollout status deployment/redis-admin --watch
+          kubectl rollout status deployment/selenium-server --watch
+          kubectl rollout status deployment/smtp-server --watch
+          kubectl rollout status deployment/webserver --watch
```

Comment:
I wouldn't necessarily wait for non-essential deployments such as redis-admin or adminer.

Reply:
Well, it depends on what purpose these lines got. I think about it as a check that everything is created correctly. Because if redis-admin, or adminer would fail on docker image or kubectl deploy, you would know it in Circle.